File: solutions/python/bob/1/bob.py
def has_letters(words):
    for c in words:
        if c.isalpha():
            return True
    return False

def response(hey_bob):
    if hey_bob.strip() == "":
        return "Fine. Be that way!"

    if has_letters(hey_bob):
        if hey_bob == hey_bob.upper():
            return "Calm down, I know what I'm doing!" if hey_bob.strip().endswith("?") else "Whoa, chill out!"
    if hey_bob.strip().endswith("?"):
        return "Sure."
    
    return "Whatever."

File: solutions/csharp/bob/1/Bob.cs
public static class Bob {
  public static string Response(string statement) {
    statement = statement.Trim();
    if (string.IsNullOrEmpty(statement)) {
      return "Fine. Be that way!";
    }
    if (statement.ToUpper() == statement && statement.Any(char.IsLetter)) {
      if (statement.EndsWith("?")) {
        return "Calm down, I know what I'm doing!";
      }
      return "Whoa, chill out!";
    }
    if (statement.EndsWith("?")) {
      return "Sure.";
    }

    return "Whatever.";
  }
}